src/stash/src/upgrade/v33_to_v34.rs

```diff
@@ -94,124 +100,140 @@ async fn migrate_cluster_replicas(tx: &mut Transaction<'_>) -> Result<(), StashE
                 value: Some(new_id),
```

Comment:
@teskje is there a reason we don't use `old_id` for system ReplicaIds? I think using the ClusterId here is incorrect.

Reply:
No, it's correct. See the comment above:
```
        // Construct the new ID format based on the plain ID and the cluster type.
        // System clusters currently always have a single replica, so we can set the replica ID to
        // the cluster ID. This does not work for user clusters.
```

This is mostly to avoid some confusion down the road. IIRC, before this change, the replica IDs were allocated like this:

cluster_id | replica_id
----------|-----------
`u1`         | `1`
`s1`         | `2`
`s2`        | `3`

So if we just prepended `u`/`s` for the existing replica_ids, we would end up with confusing system replica IDs:

cluster_id | replica_id
----------|-----------
`u1`         | `u1`
`s1`         | `s2`
`s2`        | `s3`

Instead we now have system replica_ids that match the cluster_ids.